util/src/main/java/io/kubernetes/client/util/KubeConfig.java

```diff
@@ -214,6 +222,60 @@ public String getAccessToken() {
         }
       }
     }
+    Object exec = currentUser.get("exec");
+    if (exec != null) {
+      // TODO extract to helper method for clarity
+      Map<String, Object> execMap = (Map<String, Object>) exec;
```

Comment:
It's probably better to declare an actual Java object that represents the `exec` field rather than using generic JSON accessors, it makes the code pretty ugly.

Reply:
(I realized I'm being hypocritical given the above, so feel free to ignore this...)

Reply:
> hypocritical given the above

You mean the `authProviderMap` code?

After years of bad experiences I have a general preference for plain old Java code over use of fancy reflection-based (de)serialization libraries, but if you have a strong preference for some particular approach (GSON?) I will follow your guidelines.

Reply:
@jglick - is the item "observed working with EKS (I could make a test cluster if necessary)" still open, and if not, could I help out with that?

Reply:
Yes it is still open. If you are using EKS, please help test this system.